mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/model/openjpa/AbstractJPAMailboxMessage.java

```diff
@@ -505,7 +505,10 @@ public String toString() {
     @Override
     public List<MessageAttachment> getAttachments() {
         try {
-            return new MessageParser().retrieveAttachments(getFullContent());
+            return new MessageParser().retrieveAttachments(getFullContent())
+                .stream()
+                .map(MessageAttachment.WithBytes::getMetadata)
```

Comment:
I have a question for all the classes which extends Message. 
When do we know when to return a MessageAttachment or a MessageAttachment.WithBytes ?